packages/openapi-generator/src/openapi-types.ts

```diff
@@ -24,6 +24,7 @@ export interface OpenApiOperation extends OpenAPIV3.OperationObject {
   path: string;
   requestBody?: OpenApiRequestBody;
   parameters: OpenApiParameter[];
+  'x-sap-cloud-sdk-operation-name'?: string;
```

Comment:
I think this should be done in `convertDocToUniqueOperationIds`, where we already parse the operationIds. That way you wouldn't have to touch any code down the line.

Reply:
I guess this is the major "general change".
Initially, I also think the `operation-name`/`operationId` should be treated like the `api-name`/`tags`, where we overwrite `tags` by using `api-name`.
However, from the openapi perspective, `operationId` should be unique, while there are no constraints about `tags`.
This means the example below will lead to a new spec, which is invalid, because `op1` will be used by 2 `operationId`s.
```
- api-name: apiA
  - operation-name: op1
  - operation-name: op2
- api-name: apiB
  - operation-name: op1
```
Now we disabled the validation of the openapi generator, which might not be supported in the future.
Only if when we get rid of the openapi generator and use our own solution, then we are safe.

Reply:
As discussed, we use `operationId` for now for simplicity. We'll face this when the validation is enabled.